NBitcoin/PubKey.cs

```diff
@@ -348,8 +348,9 @@ public override bool Equals(object obj)
 			PubKey item = obj as PubKey;
 			if(item == null)
 				return false;
-			return ToHex().Equals(item.ToHex());
+			return Equals(item);
 		}
+		public bool Equals(PubKey pk) => ToHex().Equals(pk.ToHex());

```

Comment:
Before this change `Equals` was comparing the PubKey hex string representation what is not great. I think this is a good opportunity for changing that and compare the internal `byte[] vch` array.

Another point here is that `public` methods should check for NULLs. The `IEquatable<PubKey>.Equals()` doesn't check it so, it can throws NRE if `myPubKey.Equals((PubKey)null)`

Reply:
Also, this is ugly. Should compare the bytes directly or underlying objects, this is less wasteful.